### ISSUE TITLE
On the fly *.sdb files generation

### ISSUE THREAD

**Issue by xarkes** [radare/radare2#7829]
For now, the *.sdb files are generated at build time with the *.sdb.txt files.
Pancake suggested to move those sdb files to the user's home and generate them at startup time if they are not already generated, or generate them when they are trying to be opened.

**Comment by lowlyw** [comment]
![img](http://xinmyx.electroduck.com/gen.php?img_url=./DEFAULT.PNG&quote=SDBs%3F%0AIn%20MY%20homedir%3F&font=./times.ttf)

i dont think shoving them into the user's homedir is a good idea. maybe in ~/.config/radare2 is ok though

**Comment by xarkes** [comment]
Yeah of course that's what I meant when I said homedir

**Comment by xarkes** [comment]
ping all @XVilka @radare @Maijin

**Comment by radare** [comment]
Skuater agrees on that change. Im still trying to think a way to do that at compile time, which is desirable to avoid unnecessary startup steps. And maybe we should do this change after the release.

**Comment by Maijin** [comment]
For sure definitely +1 since this is the reason why most of windows stuff is [XX]

**Comment by XVilka** [comment]
@xarkes placing everything in ~/.config/radare2 - sounds good.

**Comment by alvarofe** [comment]
+1 in ~/.config/radare2

**Comment by radare** [comment]
so… we want a shared library to ship all the plaintext sdbs? or load them from path is enough? imho mmaping a shlib with all the plaintext sdbs is stupid

**Comment by radare** [comment]
ok, lets do that, but not for 1.6

**Comment by xarkes** [comment]
I managed to build the sdb files with meson at build time. So no need to generate them "on the fly".

**Comment by guedou** [comment]
Being able to build SDB from regular r2 commands will be nice ! What's the best way to implement that ?

**Comment by xarkes** [comment]
As suggested by @Maijin maybe we can add a command `kc` which would compile a text file to sdb.

**Comment by Maijin** [comment]
maybe that too, but the initial one just from the bin seems more important than kc

**Comment by radare** [comment]
Yeah that would be nice. Maybe we can also implement it as a cons filter. So the output of the command would be passed to create an sdb

kc foo.sdb fk

So, if there is a k we output in kv format and the kc command will accept two args, first is the file to create and the rest will be passed to corecmd to tokenize the sdb output of the given command

How does it sounds to you?

**Comment by radare** [comment]
We will also need to reload those sdbs into r2, so maybe we need a better way to export/import in kv form, so maybe we can make a subcommand of flags, functions etc to parse a comma separated string of k=v pairs which will work as a way to import sdb into flags or so

This may be also used for sdbizing more parts of r2.

**Comment by XVilka** [comment]
@xarkes @radare what with this one?

**Comment by radare** [comment]
not for this release, so its out of my plans
